### ISSUE TITLE
API: directory { dockerBuild }

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#4015]
## Problem

The Dagger API already supports native docker build with `container { build }`. This makes for a perfectly fine DX in native SDKs, I have no complaints. **But** it is less pleasant in the context of a raw GraphQL query, because of limitations in the GraphQL language: stitching must be done manually by querying an ID in one query and copy-pasting it into the other. This is not a problem when developing "real" pipelines, since stitching is trivially available. But it is problem when using the API playground to evaluate what the Dagger API can do. Chaining a docker build from a directory would allow for much cooler demos, for example `git { directory { dockerBuild } }` for a trivial monorepo build example; etc.

## Solution

Implement `directory { dockerBuild }` as an alias to `container { build }`.

Yes, I am aware that I am contradicting my own argument for cutting `container { build }`. My worry at the time was crossing a one-way door and setting a precedent for extensions. Now that our understanding of the extension design has evolved, I am no longer worried about that (each set of extensions results in a unique graphql schema which in turn will produce a unique set of generated native clients. Whatever namespacing / extension points issues we have, they will be caught at generation and will be specific to each code generation project, which is acceptable).

TLDR: @aluzzardi you were right ;)

**Comment by shykes** [comment]
(Auto-assigning core API changes to Alex, feel free to reassign)

**Comment by shykes** [comment]
If at all possible, I would love to get this in before upcoming GraphQL announcement, since that will bring a lot of users to the playground... Not a dealbreaker but would be nice.
